libs/SmartStore/src/com/salesforce/androidsdk/smartstore/store/SmartStore.java

```diff
@@ -1020,32 +1020,27 @@ private void projectIndexedPaths(JSONObject soupElt, ContentValues contentValues
      */
     private void projectIndexedPath(JSONObject soupElt, ContentValues contentValues, IndexSpec indexSpec) {
         Object value = project(soupElt, indexSpec.path);
-        switch (indexSpec.type) {
-        case integer:
-            Long longValToUse = null;
-            try {
-                longValToUse = ((Number) value).longValue();
-            }
-            catch (Exception e) {
-                // Ignore and use the null value
-                SmartStoreLogger.e(TAG, "Unexpected error", e);
-            }
-            contentValues.put(indexSpec.columnName, longValToUse);
-            break;
-        case string:
-        case full_text:
-            contentValues.put(indexSpec.columnName, value != null ? value.toString() : null); break;
-        case floating:
-            Double doubleValToUse = null;
-            try {
-                doubleValToUse = ((Number) value).doubleValue();
-            }
-            catch (Exception e) {
-                // Ignore and use the null value
-                SmartStoreLogger.e(TAG, "Unexpected error", e);
-            }
-            contentValues.put(indexSpec.columnName, doubleValToUse); break;
-        }
+
```

Comment:
Unrelated to the bug.
More concise and less noisy projectIndexedPath method
Previous implementation would throw/log exception if integer or floating field was null